src/core/seat/keyboard.cpp

```diff
@@ -287,19 +297,13 @@ bool wf::keyboard_t::has_only_modifiers()
     return true;
 }
 
-bool wf::keyboard_t::handle_keyboard_key(uint32_t time, uint32_t key, uint32_t state)
+bool wf::keyboard_t::handle_keyboard_key(uint32_t key, uint32_t state)
 {
     using namespace std::chrono;
 
     auto& input = wf::get_core_impl().input;
     auto& seat  = wf::get_core_impl().seat;
 
-    if (wf::get_core_impl().im_relay->is_im_sent(handle))
-    {
-        mod_binding_key = 0;
```

Comment:
I am confused, isn't mod_binding_key supposed to be reset on im keys? With this change, it no longer is reset.

Reply:
I'm restoring the whole `wf::keyboard_t::handle_keyboard_key` function to previous state and moving the IM code to the `on_key` callback.

Reply:
Ok but the modifier binding key should be reset even for grabbed events. It is used to detect when a modifier binding is pressed (e.g super pressed and released shortly after). Any keys in between reset the modifier binding, grabbed or not.

Reply:
It should be reset later. It was here just because we returned early here.